### PR TITLE
Replace time with std::time, refactor Stopwatch

### PR DESCRIPTION
**Pull request by serprex**
As requested, this is #65 applied to develop

### CODE REVIEW

**Comment by ebkalderon** [comment]
@serprex I like these changes overall, especially the enum'ed Stopwatch. However, the reason why I used the `time` crate originally was because `Duration` can only return full seconds + trailing nanoseconds. To me, this seems very inconvenient for game development since it's difficult to convert that in/out of floating-point formats, especially if time values are meant to be passed into graphical shaders which often expect floats/doubles in millisecond format.

That said, I am all for using the standard library as much as possible. I would merge this in a heartbeat if `Duration` were to gain a `fn as_millis() -> f64` method. I would like to hear other members' thoughts on this. Perhaps my concerns are overblown?

**Comment by serprex** [comment]
You can always expose some basic dur_to_f32/dur_to_f64 utility functions or create a trait which when imported would add such methods to std::time::Duration

**Comment by ebkalderon** [comment]
That sounds like a good solution, yeah. I suppose those could be added to the `timing` module as well.

**Comment by serprex** [comment]
I'd prefer a discussion of such an API to be a separate PR. Unless the API is literally only ``as_millis``

**Comment by ebkalderon** [comment]
@serprex Don't know why this PR hasn't been merged yet, it looks pretty good! Unfortunately, it looks like some merge conflicts have arisen due to the removal of the `amethyst_engine` crate. Are you still working on this?

**Comment by serprex** [comment]
I'll rebase this evening

**Comment by ebkalderon** [comment]
Awesome! Thank you, and I'm sorry the PR fell through. Your help is definitely appreciated.

**Comment by serprex** [comment]
Done

File: src/engine/app.rs [hunk 1]
```diff
@@ -67,7 +68,7 @@ impl Application {
         let fixed_step = self.context.lock().unwrap().fixed_step.clone();
         let last_fixed_update = self.context.lock().unwrap().last_fixed_update.clone();
 
-        if SteadyTime::now() - last_fixed_update > fixed_step {
+        if Instant::now() - last_fixed_update > fixed_step {
             self.states.fixed_update(self.context.lock().unwrap().deref_mut());
             self.context.lock().unwrap().last_fixed_update = last_fixed_update + fixed_step;
```
Comment (ebkalderon) [inline src/engine/app.rs]: Minor tweak: with the standard library's `Duration` type, I think we can actually use the `AddAssign` operator here. I don't have access to a computer at the moment, so I can't check whether this is the case.

**Comment by ebkalderon** [comment]
Everything checks out! I'll merge this now. Thanks for the help and your patience, @serprex! :tada:

**Pull request merged**